src/decoder/lattice-faster-online-decoder.cc

```diff
@@ -989,8 +989,11 @@ void LatticeFasterOnlineDecoder::ProcessNonemitting(BaseFloat cutoff) {
   // problem did not improve overall speed.
 
   KALDI_ASSERT(queue_.empty());
-  for (const Elem *e = toks_.GetList(); e != NULL;  e = e->tail)
-    queue_.push_back(e->key);
+  for (const Elem *e = toks_.GetList(); e != NULL;  e = e->tail) {
+    StateId key = e->key;
+    if (fst_.NumInputEpsilons(key) != 0)
```

Comment:
there is another place where this check should be done, in the loop below this, before you push to the queue.  (applies to all the decoders).